pyface/ui/qt4/action/menu_manager.py

```diff
@@ -131,6 +150,23 @@ def __init__(self, manager, parent, controller):
 
         return
 
+    def _remove_event_listeners(self):
+        self._manager.on_trait_change(self.refresh, "changed", remove=True)
+        self._manager.on_trait_change(
+            self._on_enabled_changed, "enabled", remove=True
+        )
+        self._manager.on_trait_change(
+            self._on_visible_changed, "visible", remove=True
+        )
+        self._manager.on_trait_change(
+            self._on_name_changed, "name", remove=True
+        )
+        self._manager.on_trait_change(
+            self._on_image_changed, "image", remove=True
+        )
+        # Removes event listeners from downstream menu items
+        self.clear()
```

Comment:
Based on the name of this method, I would not expect `clear` to live here. Should this method be called `dispose` instead?

Reply:
Should it stay private, i.e. `_dispose`? The comment re privacy was made on another PR but I think it applies here as well.

Reply:
Right. This class already has a private name so I think we are alright to give this a public name, it would be consistent with a couple of these classes, e.g. `pyface.ui.qt4.qction.action_item._MenuItem.dispose`.